src/include/catalog/database_catalog.h

```diff
@@ -36,490 +27,233 @@ class Index;
+  /** @brief Get the OID for the specified index, or INVALID_INDEX_OID if no such INDEX exists. */
+  index_oid_t GetIndexOid(common::ManagedPointer<transaction::TransactionContext> txn, namespace_oid_t ns,
+                          const std::string &name);
+
+  /** @brief Get the storage pointer for the specified table, or nullptr if no such REGULAR_TABLE exists. */
   common::ManagedPointer<storage::SqlTable> GetTable(common::ManagedPointer<transaction::TransactionContext> txn,
                                                      table_oid_t table);
+  /** @brief Get the index pointer for the specified index, or nullptr if no such INDEX exists. */
+  common::ManagedPointer<storage::index::Index> GetIndex(common::ManagedPointer<transaction::TransactionContext> txn,
+                                                         index_oid_t index);
 
-  /**
-   * Apply a new schema to the given table.  The changes should modify the latest
-   * schema as provided by the catalog.  There is no guarantee that the OIDs for
-   * modified columns will be stable across a schema change.
-   * @param txn for the operation
-   * @param table OID of the modified table
-   * @param new_schema object describing the table after modification
-   * @return true if the operation succeeded, false otherwise
-   * @warning The catalog accessor assumes it takes ownership of the schema object
-   * that is passed.  As such, there is no guarantee that the pointer is still
-   * valid when this function returns.  If the caller needs to reference the
-   * schema object after this call, they should use the GetSchema function to
-   * obtain the authoritative schema for this table.
-   */
-  bool UpdateSchema(common::ManagedPointer<transaction::TransactionContext> txn, table_oid_t table, Schema *new_schema);
-
-  /**
-   * Get the visible schema describing the table.
-   * @param txn for the operation
-   * @param table corresponding to the requested schema
-   * @return the visible schema object for the identified table
-   */
+  /** @brief Get the schema for the specified table. */
   const Schema &GetSchema(common::ManagedPointer<transaction::TransactionContext> txn, table_oid_t table);
+  /** @brief Get the index schema for the specified index. */
+  const IndexSchema &GetIndexSchema(common::ManagedPointer<transaction::TransactionContext> txn, index_oid_t index);
 
   /**
-   * A list of all constraints on this table
-   * @param txn for the operation
-   * @param table being queried
-   * @return vector of OIDs for all of the constraints that apply to this table
+   * @brief Rename a table.
+   *
+   * @param txn         The transaction to rename the table in.
+   * @param table       The table to be renamed.
+   * @param name        The new name for the table.
+   * @return            True if the rename succeeded. False otherwise.
+   *
+   * TODO(WAN): if this logic can be pushed to PgCoreImpl, update this comment to match style
```

Comment:
So this should be primarily implemented in `PgCoreImpl` with only the DDL lock taken in the wrapper.  That said, I got a good laugh when I looked at the current implementation...

Reply:
:hankey: Will move